crates/types/src/consensus.rs

```diff
@@ -949,7 +949,7 @@ impl<TYPES: NodeType> Consensus<TYPES> {
         let mut is_leaf_extended = true;
         if let Err(e) = self.visit_leaf_ancestors(
             leaf_view,
-            Terminator::Inclusive(leaf_view - 2),
+            Terminator::Inclusive(leaf_view - 1),
```

Comment:
@lukaszrzasik In HS2 the length of the decide rule is 1 less, so I also reduced it here, are there other places in the epoch code that assume the longer chain length of the old decide rule?

Reply:
Ok I think I see what's going on, I think this makes things quite confusing though.  I thought with dual certificates we could just keep the decide rule the same.  I would strongly prefer to have no difference between decide rules and help others reach the next epoch simply by seeing proposals from the new epoch. 

For now we can try to adjust this to work with the current code but, in it's current state, it only works when you mismatch original hotstuff and this decide rule.  Basically the current code works because the decide rule we have (not hs-2) is the same as the epoch transition rule.  So if the chain should be one longer with original HS then the changing this code to `-3` shouldn't break things, but I think it does.